### ISSUE TITLE
Cache not being cleaned on planned article

### ISSUE THREAD

**Issue by acs-ferreira** [cloudflare/worker-examples#38]
Using Wordpress, the cache is not being cleaned on a planned article.

Our articles are published also on the homepage so they are published on the category but not in the homepage. only solution, re-save the article after it has been published.

**Comment by ithiru** [comment]
Yes, known issue. The plugin doesn't handle cache clear when scheduled post goes live. https://community.cloudflare.com/t/wordpress-plugin-automatic-cache-purge-doesnt-seem-to-work-when-scheduling-posts/76437

**Comment by acs-ferreira** [comment]
Does that modification solves the issue?

**Comment by ithiru** [comment]
It partially solves the problem by triggering the action, call the API to clear the cache. But the cache is not cleared properly, it should clear within 30 secs, but from my testing, it takes much more than that, even upto an hour sometimes..

Switching to https://github.com/cloudflare/worker-examples/tree/master/examples/edge-cache-html

**Comment by acs-ferreira** [comment]
Yes but I'm using the plugin of this repo and the problem remains.

And as we can see, the hook `publish_future_post` is not on the code either.

https://github.com/cloudflare/worker-examples/blob/master/examples/edge-cache-html/WordPress%20Plugin/cloudflare-page-cache/cloudflare-page-cache.php#L17

**Comment by ithiru** [comment]
yes, I also added it manually to make it work..

செவ்., 18 ஜூன், 2019, பிற்பகல் 3:05 அன்று, Carlos Ferreira <
notifications@github.com> எழுதியது:

**Comment by xtuc** [comment]
That's good to know. Would you mind sending a change to the plugin?

**Comment by utrenkner** [comment]
Any update on this issue? My client keeps asking me about planned posts...

I tried adding the hook for ``publish_future_post`` but I am not sure, which call would have the ``'x-HTML-Edge-Cache: purgeall`` header. I tried ``/wp-cron.php`` and the login page. The post went live, but did not show due to caching.

@acs-ferreira How did you get it to work (half way)? What did you change in the code and which URL did you call to have the purgeall header appear in the response?
